armory/scenarios/image_classification.py

```diff
@@ -193,7 +209,22 @@ def _evaluate(
                     elif targeted:
                         y_target = label_targeter.generate(y)
                         generate_kwargs["y"] = y_target
-                    x_adv = attack.generate(x=x, **generate_kwargs)
+
+                    if skip_misclassified:
+                        batch_size = x.shape[0]
+                        if batch_size > 1:
+                            logger.warning(
+                                "Ignoring --skip-misclassified flag since batch_size is "
+                                "greater than 1."
+                            )
+                            x_adv = attack.generate(x=x, **generate_kwargs)
+                        else:
+                            if benign_acc[batch_idx] == 0:
+                                x_adv = x
+                            else:
+                                x_adv = attack.generate(x=x, **generate_kwargs)
+                    else:
+                        x_adv = attack.generate(x=x, **generate_kwargs)
```

Comment:
This section could be simplified (once the warning/error is moved) to 
```
if skip_misclassified and benign_acc[batch_idx] == 0:
    x_adv = x
else:
    x_adv = attack.generate(x=x, **generate_kwargs)

Reply:
agreed, done